deploy/examples/create-external-cluster-resources.py

```diff
@@ -556,23 +556,30 @@ def endpoint_dial(self, endpoint_str, timeout=3, cert=None):
         if isinstance(self.cluster, DummyRados):
             return
         protocols = ["http", "https"]
+        response_error = None
         for prefix in protocols:
             try:
                 ep = "{}://{}".format(prefix, endpoint_str)
+                verify = None
                 # If verify is set to a path to a directory,
                 # the directory must have been processed using the c_rehash utility supplied with OpenSSL.
                 if prefix == "https" and cert and self._arg_parser.rgw_skip_tls:
+                    verify = False
                     r = requests.head(ep, timeout=timeout, verify=False)
                 elif prefix == "https" and cert:
+                    verify = cert
                     r = requests.head(ep, timeout=timeout, verify=cert)
                 else:
                     r = requests.head(ep, timeout=timeout)
                 if r.status_code == 200:
-                    return prefix
+                    return prefix, verify
             except:
+                response_error = r
                 continue
```

Comment:
```suggestion
            except Exception as response_error:
                continue
```
(Or probably better handle specific exceptions instead of the common `Exception`.)

Reply:
The problem was r was not defined outside the for loop

Reply:
I thought, that you want to have the actual failure, which caused the raise of the exception, saved in to the `response_error` - thus you have to catch the exception and save it to variable.

Reply:
@dahorak just a gentle reminder if you are okay with the changes can you test them on the respective cluster